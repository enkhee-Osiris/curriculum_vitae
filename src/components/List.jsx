import { v4 as uuidv4 } from 'uuid';

function List({ items }) {
  return (
    <ul className="space-y-1 list-disc list-inside">
      {items.map((item) => (
        <li key={uuidv4()}>{item}</li>
      ))}
    </ul>
  );
}

export default List;
