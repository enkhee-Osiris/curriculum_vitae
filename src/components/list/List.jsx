import React from 'react';
import { arrayOf, string } from 'prop-types';

const List = ({ items }) => {
  /* eslint-disable react/no-array-index-key */
  const listItems = items.map((item, index) => (
    <li key={index}>
      {item}
    </li>));

  return <ul>{listItems}</ul>;
};

List.propTypes = {
  items: arrayOf(string).isRequired,
};

export { List as default };
