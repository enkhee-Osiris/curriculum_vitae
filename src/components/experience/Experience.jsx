import React, { Fragment } from 'react';
import { VerticalTimeline, VerticalTimelineElement } from 'react-vertical-timeline-component';
import { arrayOf, shape, string, number } from 'prop-types';
import 'react-vertical-timeline-component/style.min.css';
import List from '../list/List';

const Experience = ({ datas }) => {
  const timelineElements = datas.map(data => (
    <VerticalTimelineElement
      key={data.id}
      className="vertical-timeline-element--work"
      date={data.date}
    >
      <h4 className="vertical-timeline-element-title">{data.position}</h4>
      <h5 className="vertical-timeline-element-subtitle">{data.company}</h5>
      <List items={data.items} />
    </VerticalTimelineElement>
  ));
  return (
    <Fragment>
      <h3>Experience</h3>
      <VerticalTimeline animate={false}>
        {timelineElements}
      </VerticalTimeline>
    </Fragment>
  );
};

Experience.propTypes = {
  datas: arrayOf(shape({
    id: number,
    position: string,
    company: string,
    items: arrayOf(string),
    date: string,
  })).isRequired,
};

export { Experience as default };
