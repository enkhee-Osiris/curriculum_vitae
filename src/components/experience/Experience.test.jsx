import React from 'react';
import { shallow } from 'enzyme';
import { VerticalTimeline, VerticalTimelineElement } from 'react-vertical-timeline-component';
import Experience from './Experience';
import List from '../list/List';

describe('<Experience />', () => {
  const props = {
    datas: [
      {
        id: 1,
        position: 'test position',
        company: 'test comp',
        items: [
          'test item',
        ],
        date: 'April 2017 - present',
      },
    ],
  };

  it('should be defined', () => {
    expect(Experience).toBeDefined();
  });

  it('renders correctly', () => {
    const experience = shallow(<Experience {...props} />);

    expect(experience).toMatchSnapshot();
  });

  it('renders h3', () => {
    const experience = shallow(<Experience {...props} />);
    const title = <h3>Experience</h3>;

    expect(experience.contains(title)).toEqual(true);
  });

  it('renders VerticalTimeline', () => {
    const experience = shallow(<Experience {...props} />);

    expect(experience.find(VerticalTimeline).length).toBeGreaterThan(0);
  });

  it('renders VerticalTimelineElement', () => {
    const experience = shallow(<Experience {...props} />);

    expect(experience.find(VerticalTimelineElement).length).toBeGreaterThan(0);
  });

  it('renders List', () => {
    const experience = shallow(<Experience {...props} />);

    expect(experience.find(List).length).toBeGreaterThan(0);
  });

  it('renders timeline datas', () => {
    const experience = shallow(<Experience {...props} />);

    expect(experience.find('h4').first().text()).toEqual(props.datas[0].position);
    expect(experience.find('h5').first().text()).toEqual(props.datas[0].company);
  });
});
