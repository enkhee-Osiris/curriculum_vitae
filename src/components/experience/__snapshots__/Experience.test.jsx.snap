// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Experience /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Experience
    datas={
      Array [
        Object {
          "company": "test comp",
          "date": "April 2017 - present",
          "id": 1,
          "items": Array [
            "test item",
          ],
          "position": "test position",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Experience
        </h3>,
        <VerticalTimeline
          animate={false}
          className=""
          layout="2-columns"
        >
          <VerticalTimelineElement
            className="vertical-timeline-element--work"
            date="April 2017 - present"
            icon={null}
            iconOnClick={null}
            iconStyle={null}
            id=""
            position=""
            style={null}
            visibilitySensorProps={
              Object {
                "offset": Object {
                  "bottom": 80,
                },
                "partialVisibility": true,
              }
            }
          >
            <h4
              className="vertical-timeline-element-title"
            >
              test position
            </h4>
            <h5
              className="vertical-timeline-element-subtitle"
            >
              test comp
            </h5>
            <List
              items={
                Array [
                  "test item",
                ]
              }
            />
          </VerticalTimelineElement>
        </VerticalTimeline>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Experience",
        },
        "ref": null,
        "rendered": "Experience",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "animate": false,
          "children": Array [
            <VerticalTimelineElement
              className="vertical-timeline-element--work"
              date="April 2017 - present"
              icon={null}
              iconOnClick={null}
              iconStyle={null}
              id=""
              position=""
              style={null}
              visibilitySensorProps={
                Object {
                  "offset": Object {
                    "bottom": 80,
                  },
                  "partialVisibility": true,
                }
              }
            >
              <h4
                className="vertical-timeline-element-title"
              >
                test position
              </h4>
              <h5
                className="vertical-timeline-element-subtitle"
              >
                test comp
              </h5>
              <List
                items={
                  Array [
                    "test item",
                  ]
                }
              />
            </VerticalTimelineElement>,
          ],
          "className": "",
          "layout": "2-columns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h4
                  className="vertical-timeline-element-title"
                >
                  test position
                </h4>,
                <h5
                  className="vertical-timeline-element-subtitle"
                >
                  test comp
                </h5>,
                <List
                  items={
                    Array [
                      "test item",
                    ]
                  }
                />,
              ],
              "className": "vertical-timeline-element--work",
              "date": "April 2017 - present",
              "icon": null,
              "iconOnClick": null,
              "iconStyle": null,
              "id": "",
              "position": "",
              "style": null,
              "visibilitySensorProps": Object {
                "offset": Object {
                  "bottom": 80,
                },
                "partialVisibility": true,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test position",
                  "className": "vertical-timeline-element-title",
                },
                "ref": null,
                "rendered": "test position",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test comp",
                  "className": "vertical-timeline-element-subtitle",
                },
                "ref": null,
                "rendered": "test comp",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "items": Array [
                    "test item",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Experience
          </h3>,
          <VerticalTimeline
            animate={false}
            className=""
            layout="2-columns"
          >
            <VerticalTimelineElement
              className="vertical-timeline-element--work"
              date="April 2017 - present"
              icon={null}
              iconOnClick={null}
              iconStyle={null}
              id=""
              position=""
              style={null}
              visibilitySensorProps={
                Object {
                  "offset": Object {
                    "bottom": 80,
                  },
                  "partialVisibility": true,
                }
              }
            >
              <h4
                className="vertical-timeline-element-title"
              >
                test position
              </h4>
              <h5
                className="vertical-timeline-element-subtitle"
              >
                test comp
              </h5>
              <List
                items={
                  Array [
                    "test item",
                  ]
                }
              />
            </VerticalTimelineElement>
          </VerticalTimeline>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Experience",
          },
          "ref": null,
          "rendered": "Experience",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "animate": false,
            "children": Array [
              <VerticalTimelineElement
                className="vertical-timeline-element--work"
                date="April 2017 - present"
                icon={null}
                iconOnClick={null}
                iconStyle={null}
                id=""
                position=""
                style={null}
                visibilitySensorProps={
                  Object {
                    "offset": Object {
                      "bottom": 80,
                    },
                    "partialVisibility": true,
                  }
                }
              >
                <h4
                  className="vertical-timeline-element-title"
                >
                  test position
                </h4>
                <h5
                  className="vertical-timeline-element-subtitle"
                >
                  test comp
                </h5>
                <List
                  items={
                    Array [
                      "test item",
                    ]
                  }
                />
              </VerticalTimelineElement>,
            ],
            "className": "",
            "layout": "2-columns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <h4
                    className="vertical-timeline-element-title"
                  >
                    test position
                  </h4>,
                  <h5
                    className="vertical-timeline-element-subtitle"
                  >
                    test comp
                  </h5>,
                  <List
                    items={
                      Array [
                        "test item",
                      ]
                    }
                  />,
                ],
                "className": "vertical-timeline-element--work",
                "date": "April 2017 - present",
                "icon": null,
                "iconOnClick": null,
                "iconStyle": null,
                "id": "",
                "position": "",
                "style": null,
                "visibilitySensorProps": Object {
                  "offset": Object {
                    "bottom": 80,
                  },
                  "partialVisibility": true,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test position",
                    "className": "vertical-timeline-element-title",
                  },
                  "ref": null,
                  "rendered": "test position",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test comp",
                    "className": "vertical-timeline-element-subtitle",
                  },
                  "ref": null,
                  "rendered": "test comp",
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "items": Array [
                      "test item",
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
